pyvo/dal/tap.py

```diff
@@ -870,15 +870,23 @@ def results(self):
     @property
     def result(self):
         """
-        Returns the UWS result with id='result' if it exists, otherwise None.
+        Returns the UWS result that corresponds to the standard TAP result
+        endpoint: "results/result".
+
+        If no such result is found it falls back to the old behavior of returning
+        the first result with id 'result'.
         """
         try:
             for r in self._job.results:
-                if r.id_ == 'result':
+                if r.href and r.href.endswith("results/result"):
+                    return r
+
+            for r in self._job.results:
+                if r.href and r.href.strip() and r.id_ == 'result':
```

Comment:
Isn't this supposed to be the fallback for the same behaviour as before? If yes, lose the first two conditions.

Reply:
Yes indeed this is not 100% backwards compatible, however I think the previous code allowed a job that specified a result with an empty `href` so i figured it is more correct to return `None` in that case. (test-case: `test_job_result_empty_href`)

However if we don't want this to be the case I'm happy to remove the first two conditions and leave that part as before.

Reply:
Let's wait for Markus' review here.